one-ui/src/main/java/com/marklogic/hub/oneui/models/HubConfigSession.java

```diff
@@ -1000,14 +1002,14 @@ public void afterPropertiesSet() throws Exception {
         if (hubConfigImpl.getAdminManager() == null) {
             hubConfigImpl.setAdminManager(new AdminManager());
         }
+        String projectDirectory = environmentService.getProjectDirectory();
         if (hubConfigImpl.getAppConfig() == null) {
-            hubConfigImpl.setAppConfig(new AppConfig(), true);
+            hubConfigImpl.setAppConfig(new AppConfig(Paths.get(projectDirectory).toFile()), true);
         }
-        hubConfigImpl.createProject(environmentService.getProjectDirectory());
+        hubConfigImpl.createProject(projectDirectory);
```

Comment:
I think we really need unit tests for the changes in here. My experience over the past couple months with trying to make changes to HubConfigImpl is that it's very difficult to know what any change might break, due to the class doing too many things. You can likely pull some of these changes into protected methods so that it's easier to test what they're doing.